connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/v2/MsSqlServerIntegrationSuite.scala

```diff
@@ -146,4 +146,66 @@ class MsSqlServerIntegrationSuite extends DockerJDBCIntegrationV2Suite with V2JD
         |""".stripMargin)
     assert(df.collect().length == 2)
   }
+
+  test("SPARK-50087: SqlServer handle booleans in IF in SELECT test") {
+    // This doesn't compile on SqlServer unless result boolean expressions
+    // in IF / CASE WHEN are wrapped with an IIF(<>, 1, 0).
+    val df = sql(
+      s"""|WITH dummy AS (
+          |  SELECT
+          |    DISTINCT name AS full_name,
+          |    UPPER(name) AS test_type,
+          |    name,
+          |    IF(
+          |      LOWER(name) = 'adfsaef' OR LOWER(name) = 'agadg',
+          |      'agfagff',
+          |      IF(
+          |        LOWER(name) = 'adgfda' OR LOWER(name) = 'ssadf',
+          |        'sxzvfvxf',
+          |        IF(
+          |          LOWER(name) = 'sdfadsf' OR LOWER(name) = 'sadfgvad',
+          |          'sAFvadsfvcds',
+          |          LOWER(name)
+          |        )
+          |      )
+          |    ) AS test_type_name
+          |  FROM $catalogName.employee
+          |),
+          |dummy_new AS (
+          |  SELECT *
+          |  FROM dummy WHERE test_type_name = 'safcdfz'
+          |)
+          |SELECT * FROM dummy_new limit 1""".stripMargin
+    )
+    df.collect()
```

Comment:
should we have some pushdown check ?

Reply:
I'll add an assert with External query check